indra/literature/elsevier_client.py

```diff
@@ -39,55 +39,72 @@
+        def check_api_keys(*args, **kwargs):
+            api_key_env_name = 'ELSEVIER_API_KEY'
+            inst_key_env_name = 'ELSEVIER_INST_KEY'
+            global ELSEVIER_KEYS
+            if ELSEVIER_KEYS is None:
+                ELSEVIER_KEYS = {}
+                # THE API KEY IS NOT UNDER VERSION CONTROL FOR SECURITY
+                # For more information see http://dev.elsevier.com/
 
-if not has_config(api_key_env_name):
-    logger.error('API key ' + api_key_env_name + ' not found in ' + \
-                 'configuration file or environment variable.')
-if has_config(api_key_env_name):
-    elsevier_keys['X-ELS-APIKey'] = get_config(api_key_env_name)
+                # Using this file is deprecated.
+                # global API_KEY_FILE
+                # path_to_here = os.path.dirname(os.path.realpath(__file__))
+                # API_KEY_FILE = os.path.join(path_to_here, 'elsevier_api_keys')
 
+                # Try to read in Elsevier API keys. For each key, first check
+                # the environment variables, then check the INDRA config file.
+                if not has_config(inst_key_env_name):
+                    logger.warning('Institution API key %s not found in config '
+                                   'file or environment variable: this will '
+                                   'limit ability to %s'
+                                   % (inst_key_env_name, task_desc))
+                ELSEVIER_KEYS['X-ELS-Insttoken'] = get_config(inst_key_env_name)
 
+                if not has_config(api_key_env_name):
+                    logger.error('API key %s not found in configuration file '
+                                 'or environment variable: cannot %s'
+                                 % (api_key_env_name, task_desc))
+                    return failure_ret
+                ELSEVIER_KEYS['X-ELS-APIKey'] = get_config(api_key_env_name)
+            elif 'X-ELS-APIKey' not in ELSEVIER_KEYS.keys():
+                logger.error('No Elsevier API key %s found: cannot %s'
+                             % (api_key_env_name, task_desc))
+                return failure_ret
+            return func(*args, **kwargs)
+        return check_api_keys
+    return check_func_wrapper
+
+
+@ensure_api_keys('check article entitlement', False)
 def check_entitlement(doi):
-    if elsevier_keys is None:
-        logger.error('Missing API key, could not check article entitlement.')
-        return False
     if doi.lower().startswith('doi:'):
         doi = doi[4:]
     url = '%s/%s' % (elsevier_entitlement_url, doi)
     params = {'httpAccept': 'text/xml'}
-    res = requests.get(url, params, headers=elsevier_keys)
+    res = requests.get(url, params, headers=ELSEVIER_KEYS)
     if not res.status_code == 200:
         logger.error('Could not check entitlements for article %s: '
                      'status code %d' % (doi, res.status_code))
         logger.error('Response content: %s' % res.text)
         return False
```

Comment:
Yeah, this doesn't seem to do much...I left a todo there, because I'm not really sure what exactly it's intent is. It's not used anywhere that I or PyCharm can find.

Reply:
I talked with Jon and figured it out.